Setup/UpgradeData.php

```diff
@@ -0,0 +1,103 @@
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @author Nosto Solutions Ltd <contact@nosto.com>
+ * @copyright 2017 Nosto Solutions Ltd
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD 3-Clause
+ *
+ */
+
+namespace Nosto\Tagging\Setup;
+
+use Magento\Framework\Setup\ModuleContextInterface;
+use Magento\Framework\Setup\ModuleDataSetupInterface;
+use Magento\Framework\Setup\UpgradeDataInterface;
+use Nosto\Tagging\Helper\Account as NostoHelperAccount;
+use Nosto\Tagging\Helper\Url as NostoHelperUrl;
+use Magento\Framework\App\Config\Storage\WriterInterface;
+use Magento\Store\Model\ScopeInterface;
+
+class UpgradeData implements UpgradeDataInterface
+{
+    /** @var NostoHelperAccount */
+    private $nostoHelperAccount;
+
+    /** @var NostoHelperUrl */
+    private $nostoHelperUrl;
+
+    /** @var WriterInterface */
+    private $config;
+
+    /**
+     * UpgradeData constructor.
+     * @param NostoHelperAccount $nostoHelperAccount
+     * @param NostoHelperUrl $nostoHelperUrl
+     * @param WriterInterface $appConfig
+     */
+    public function __construct(
+        NostoHelperAccount $nostoHelperAccount,
+        NostoHelperUrl $nostoHelperUrl,
+        WriterInterface $appConfig
+    ) {
+        $this->nostoHelperAccount = $nostoHelperAccount;
+        $this->nostoHelperUrl = $nostoHelperUrl;
+        $this->config = $appConfig;
+    }
+
+    /**
+     * @param ModuleDataSetupInterface $setup
+     * @param ModuleContextInterface $context
+     */
+    public function upgrade(ModuleDataSetupInterface $setup, ModuleContextInterface $context) // @codingStandardsIgnoreLine
+    {
+        if (version_compare($context->getVersion(), '3.1.0', '>=')) {
+            $this->insertStoreDomain();
+        }
+    }
+
+    /**
+     * Insert store domain when missing in database
+     */
+    private function insertStoreDomain()
+    {
+        $stores = $this->nostoHelperAccount->getStoresWithNosto();
+        foreach ($stores as $store) {
+            if ($this->nostoHelperAccount->getStoreFrontDomain($store) === null) {
+                // @codingStandardsIgnoreLine
```

Comment:
Why are you ignoring this?

Reply:
It would give this error:
<error line="93" column="32" severity="warning" message="Model LSD method save() detected in loop" source="MEQP1.Performance.Loop.ModelLSD"/>

Reply:
👍